File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String str) {
        char[] chars = str.toCharArray();
        Stack<Character> st = new Stack<>();
        for(char c : chars){
            if(st.isEmpty() || !(st.peek() == '(' && c == ')')){
                st.push(c);
            } else {
                st.pop();
            }
        }
        return st.isEmpty();
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    const stack = [];
    for(let char of s){
        if(char === '('){
            stack.push(char);
        }else if(char === ')'){
            if(stack.length === 0) return false;
            else stack.pop();
        }
    }
    return stack.length > 0 ? false : true;
}